examples/reporting/campaign_report_to_csv.rb

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/ruby
+# Encoding: utf-8
+#
+# Copyright:: Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This code example illustrates how to get metrics about a campaign and 
+# serialize the result as a CSV file.
+
+require 'optparse'
+require 'time'
+require 'csv'
+require 'google/ads/google_ads'
+
+PAGE_SIZE = 1000
+
+def result_row_as_csv_hash(result_row)
+  {
+    id: result_row.campaign.id.value,
+    name: result_row.campaign.name.value,
+    date: Time.parse(result_row.segments.date.value),
+    impressions: result_row.metrics.impressions.value,
+    clicks: result_row.metrics.clicks.value,
+    cost_micros: result_row.metrics.cost_micros.value,
+  }
+end
+
+def write_campaign_report_csv(customer_id, target_filepath)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  ga_service = client.service(:GoogleAds)
+
+  query = <<~QUERY
+      SELECT campaign.id,
+             campaign.name,
+             segments.date,
+             metrics.impressions,
+             metrics.clicks,
+             metrics.cost_micros
+      FROM campaign
+      WHERE segments.date DURING LAST_7_DAYS
+        AND campaign.status = 'ENABLED'
+      ORDER BY segments.date DESC
+  QUERY
+
+  response = ga_service.search(customer_id, query, page_size: PAGE_SIZE)
+  # convert the Google Ads response rows in to CSV ready hash objects
+  csv_rows = response.map { |row| result_row_as_csv_hash(row) }
+
+  CSV.open(target_filepath, "wb") do |csv|
+    # use the keys of the first csv_row as a header row
+    csv << csv_rows.first.keys
```

Comment:
I think we may want to output the entire path to the field (i.e. what you specify in the select clause) rather than just the keys you use in the hash.